python/tk_framework_desktopserver/server_protocol.py

```diff
@@ -123,45 +126,81 @@ def onMessage(self, payload, isBinary):
         message_host = MessageHost(self, message)
 
         # Check protocol version
-        if message["protocol_version"] != self.PROTOCOL_VERSION:
-            message_host.report_error("Error. Wrong protocol version [%s] " % self.PROTOCOL_VERSION)
+        if message["protocol_version"] not in self.SUPPORTED_PROTOCOL_VERSIONS:
+            message_host.report_error(
+                "Unsupported protocol version: %s " % message["protocol_version"]
+            )
             return
 
-        # Run each request from a thread, even tough it might be something very simple like opening a file. This
+        # TODO: This isn't going to work if we have two different clients
+        # connecting at two different protocol versions. This is a temp
```

Comment:
Looking at this again, I believe it's just a matter of storing the protocol version on the object instead of the class. Each unique wss connection from a client will spin up its own protocol handler object, so in theory each connection could come from a different protocol version and work just fine.